### PR TITLE
ROU-3106: Arrange Data not working properly when using complex struct…

### PR DESCRIPTION
**Pull request by gnbm**
Fix an issue when handling more complex structures while obtaining the metadata for the generated JSON from the ArrangeData method.

### What was happening
* During a testing session, we found out that the Data Grid was throwing errors when the resulting JSON comes from a more complex structure like a structure with entities as an attribute
* We found out that the issues are inside the ArranjeData and that in these scenarios we’re not removing the prefix “ss“

### What was done
* Changed the method _cleanRecName_ to handle more complex structures when obtaining the metadata

### Test Steps
1. Create a screen with a grid with a data source from an advanced query that is appended to a list with a structure where we have an entity and another structure inside
2. Check that the information is loaded successfully
3. Click in **Deserialize**
4. Check that the data is correct and that the product name "My Personal Desktop" is displayed on a feedback message

### Screenshots
ROU3106_Preview

### Checklist
* [X] tested locally
* [X] documented the code
* [X] clean all warnings and errors of eslint

### CODE REVIEW

File: code/extension/DataGridUtils/Source/NET/obj/Release/DataGridUtils.csproj.FileListAbsolute.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+C:\Users\gna\Desktop\DataGridUtils\Source\NET\bin\OutSystems.NssDataGridUtils.dll
```
Comment (glundgren93) [inline code/extension/DataGridUtils/Source/NET/obj/Release/DataGridUtils.csproj.FileListAbsolute.txt:1]: I don't think we should be committing files from obj here. What do you think?

**Review by glundgren93 — APPROVED** [review]

File: code/extension/DataGridUtils/Source/NET/DataGridUtils.cs [hunk 2]
```diff
@@ -1,7 +1,7 @@
 
-namespace OutSystems.NssDataGridDataConvert {
+namespace OutSystems.NssDataGridUtils {
 
-	public class CssDataGridDataConvert: IssDataGridDataConvert {
+	public class CssDataGridUtils: IssDataGridUtils {
```
Comment (rugoncalves) [inline code/extension/DataGridUtils/Source/NET/DataGridUtils.cs:4]: Nice detail!

File: code/extension/DataGridUtils/Templates/NET/Interface.cs [hunk 3]
```diff
@@ -22,6 +22,6 @@ public interface IssDataGridDataConvert {
 		/// <param name="ssDataJSON">JSON with the data and metadata for the data grid</param>
 		void MssConvertData2JSON(object ssData, out string ssDataJSON);
```
Comment (rugoncalves) [inline code/extension/DataGridUtils/Templates/NET/Interface.cs:23]: Let's remove this method if it's deprecated? The extension is only used internally.

**Review by rugoncalves — APPROVED** [review]
Looks good - would just propose to remove the deprecated method.

File: code/extension/DataGridUtils/Templates/NET/Interface.cs (hunk 3 above)
Comment (gnbm) [reply code/extension/DataGridUtils/Templates/NET/Interface.cs:23]: will do that on the next release

**Pull request merged**